src/ring_theory/ideal/basic.lean

```diff
@@ -373,6 +375,56 @@ theorem is_prime.pow_mem_iff_mem {I : ideal α} (hI : I.is_prime)
   {r : α} (n : ℕ) (hn : 0 < n) : r ^ n ∈ I ↔ r ∈ I :=
 ⟨hI.mem_of_pow_mem n, (λ hr, I.pow_mem_of_mem hr n hn)⟩
 
+theorem pow_sum_mem_span_pow (s : multiset α) (n : ℕ) :
+  s.sum ^ (s.card * n + 1) ∈ span ((s.map (λ x, x ^ (n + 1))).to_finset : set α) :=
```

Comment:
The mixture of finset, multiset, and set in this lemma statement feels very strange to me. I haven't looked at this long enough to get a feel for what a simpler statement might look like.

Reply:
Is there a better way to turn a `multiset` into a `set`?

Reply:
Here's the version of the lemma that avoids `multiset` entirely:
```
theorem pow_sum_mem_span_pow {ι} (s : finset ι) (f : ι → α) (n : ℕ) :
  (∑ i in s, f i) ^ (s.card * n + 1) ∈ span (s.image (λ i, f i ^ (n + 1)) : set α) :=
begin
  apply s.induction_on,
  { simp },
  rintros i s hi hs,
  simp only [finset.coe_insert, finset.card_insert_of_not_mem hi, add_pow, finset.image_insert,
    finset.sum_insert hi],
  refine submodule.sum_mem _ _,
  intros c hc,
  rw mem_span_insert,
  by_cases h : n+1 ≤ c,
  { refine ⟨f i ^ (c - (n + 1)) * (∑ i in s, f i) ^ ((s.card + 1) * n + 1 - c) *
      (((s.card + 1) * n + 1).choose c), 0, submodule.zero_mem _, _⟩,
    rw mul_comm _ (f i ^ (n + 1)),
    simp_rw ← mul_assoc,
    rw [← pow_add, add_zero, add_tsub_cancel_of_le h], },
  { use 0,
    simp_rw [zero_mul, zero_add],
    refine ⟨_,_,rfl⟩,
    replace h : c ≤ n := nat.lt_succ_iff.mp (not_le.mp h),
    have : (s.card + 1) * n + 1 - c = s.card * n + 1 + (n - c),
    { rw [add_mul, one_mul, add_assoc, add_comm n 1, ← add_assoc, add_tsub_assoc_of_le h] },
    rw [this, pow_add],
    simp_rw [mul_assoc, mul_comm ((∑ i in s, f i) ^ (s.card * n + 1)), ← mul_assoc],
    exact mul_mem_left _ _ hs }
end
```
I'm not sure whether it's easier to prove the multiset one from this one or vice versa.

Reply:
Ah, it's easier to start with your lemma and prove the more readable finset result:
```lean
theorem pow_sum_mem_span_pow {ι} (s : finset ι) (f : ι → α) (n : ℕ) :
  (∑ i in s, f i) ^ (s.card * n + 1) ∈ span (s.image (λ i, f i ^ (n + 1)) : set α) :=
begin
  convert pow_multiset_sum_mem_span_pow (s.1.map f) n,
  { rw multiset.card_map, refl },
  rw [multiset.map_map, multiset.to_finset_map, finset.val_to_finset],
end
```
Can you add that version and rename the existing one to `pow_multiset_sum_mem_span_pow`?